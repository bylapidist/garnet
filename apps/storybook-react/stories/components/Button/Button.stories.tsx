import * as React from 'react';
import { Button } from '@lapidist/garnet-react';
import type { ButtonProps } from '@lapidist/garnet-react';
import type { Meta, StoryObj, StoryFn } from '@storybook/react';

export default {
    title: 'Button',
    component: Button,
} as Meta;

const Template: StoryFn<ButtonProps> = (args) => <Button {...args} />;
export const DefaultButton: StoryObj<ButtonProps> = Template.bind({});
DefaultButton.args = {
    type: 'button',
};
