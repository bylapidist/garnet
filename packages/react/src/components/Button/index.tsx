import React from 'react';

export interface ButtonProps {
    type: 'button' | 'submit';
}

const Button = (props: ButtonProps) => {
    return <button type={props.type}>hi</button>;
};

export default Button;
