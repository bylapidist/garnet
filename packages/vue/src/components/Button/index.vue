<script setup lang="ts">
    import { ButtonProps } from '@lapidist/garnet-types';

    defineProps<ButtonProps>()
</script>

<template>
    <button :type="type">hi</button>
</template>
