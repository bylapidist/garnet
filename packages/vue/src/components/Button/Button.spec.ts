import { mount } from '@vue/test-utils';
import { describe, expect, test } from 'vitest';
import Button from './index.vue';

describe('Button', () => {
    test('it matches snapshot', () => {
        const wrapper = mount(Button, { props: { type: 'button' } });
        expect(wrapper.html()).toMatchSnapshot();
    });
});
