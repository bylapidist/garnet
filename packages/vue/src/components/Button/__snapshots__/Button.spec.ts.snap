// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > it matches snapshot 1`] = `"<button type=\\"button\\">hi</button>"`;
